core/src/tile/tileWorker.h

```diff
@@ -5,42 +5,57 @@
 #include <condition_variable>
 #include <thread>
 #include <mutex>
+#include <atomic>
 
 namespace Tangram {
 
-class TileManager;
 class DataSource;
+class Scene;
 class Tile;
 class TileTask;
 class View;
 
 class TileWorker {
-    
+
 public:
-    
-    TileWorker(TileManager& _tileManager, int _num_worker);
+
+    TileWorker(int _num_worker);
 
     ~TileWorker();
-    
+
     void enqueue(std::shared_ptr<TileTask>&& task);
-    
+
     void stop();
-    
+
     bool isRunning() const { return m_running; }
-    
+
+    // Check pending-tiles flag. Resets flag on each call..
+    // TODO better name checkAndResetPendingTilesFlag?
+    bool checkPendingTiles() {
+        if (m_pendingTiles) {
+            m_pendingTiles = false;
+            return true;
+        }
+        return false;
+    }
+
+    void setScene(std::shared_ptr<Scene> _scene) { m_scene = _scene; }
+
 private:
 
     void run();
-    
-    TileManager& m_tileManager;
-    
+
     bool m_running;
-    
+
+    std::atomic<bool> m_pendingTiles;
```

Comment:
I think atomic<bool> is appropriate here. atomic_flag seems to be useful if one wants to implement a spinlock